src/csv_processor.cpp

```diff
@@ -1,25 +1,268 @@
 #include "csv_processor.hpp"
 
-#include "csv_arrow_ingest.hpp"
+#include "decryption.hpp"
 #include "duckdb.hpp"
+#include "ingest_properties.hpp"
 #include "md_logging.hpp"
+#include "memory_backed_file.hpp"
+#include "schema_types.hpp"
 
-#include <arrow/c/bridge.h>
 #include <fstream>
 #include <functional>
+#include <limits>
 #include <memory>
+#include <optional>
 #include <stdexcept>
 #include <string>
+#include <unistd.h>
+
+namespace fs = std::filesystem;
 
 namespace {
+
 void validate_file(const std::string &file_path) {
-  std::ifstream fs(file_path.c_str());
-  if (fs.good()) {
-    fs.close();
+  const std::ifstream fs(file_path);
+  if (fs.fail()) {
+    throw std::system_error(errno, std::generic_category(),
+                            "Failed to open file <" + file_path + ">");
+  }
+}
+
+MemoryBackedFile
+create_file_with_decrypted_content(const std::string &encrypted_file_path,
```

Comment:
maybe decrypt_file_into_memory? This name somewhat implies that a physical file is created, which is not the case on Linux.